src/hotspot/cpu/aarch64/aarch64_vector_ad.m4

```diff
@@ -0,0 +1,4693 @@
+    __ sve_index($dst$$FloatRegister, __ D, -16, 1);
+    __ sve_cmp(Assembler::EQ, $pgtmp$$PRegister, __ D, ptrue,
+               $dst$$FloatRegister, (int)($idx$$constant) - 16);
+    __ sve_orr($dst$$FloatRegister, $src$$FloatRegister, $src$$FloatRegister);
+    __ sve_cpy($dst$$FloatRegister, __ D, $pgtmp$$PRegister, $val$$FloatRegister);
+  %}
+  ins_pipe(pipe_slow);
+%}
+
+// ------------------------------ Extract --------------------------------------
+dnl
+dnl EXTRACT_INT_SMALL($1,   $2,    $3,       $4,   $5  )
+dnl EXTRACT_INT_SMALL(type, index, arg_type, insn, size)
+define(`EXTRACT_INT_SMALL', `
+instruct extract$1_index_lt$2($3 dst, vReg src, immI idx) %{
+  predicate(n->in(2)->get_int() < $2);
+  match(Set dst (Extract$1 src idx));
+  format %{ "extract$1_index_lt$2 $dst, $src, $idx\t# index < $2" %}
+  ins_encode %{
+    __ $4($dst$$Register, $src$$FloatRegister, __ $5, (int)($idx$$constant));
+  %}
+  ins_pipe(pipe_slow);
+%}')dnl
+dnl
+dnl
+dnl EXTRACT_INT_LARGE($1,   $2,    $3,       $4       )
+dnl EXTRACT_INT_LARGE(type, index, arg_type, data_type)
+define(`EXTRACT_INT_LARGE', `
+instruct extract$1_index_ge$2($3 dst, vReg src, immI idx, vReg tmp) %{
+  predicate(n->in(2)->get_int() >= $2);
+  match(Set dst (Extract$1 src idx));
+  effect(TEMP tmp);
+  format %{ "extract$1_index_ge$2 $dst, $src, $idx\t# index >=$2. KILL $tmp" %}
+  ins_encode %{
+    assert(UseSVE > 0, "must be sve");
+    __ sve_extract_integral($dst$$Register, $4, $src$$FloatRegister,
+                            (int)($idx$$constant), $tmp$$FloatRegister);
+  %}
+  ins_pipe(pipe_slow);
+%}')dnl
+dnl
+// BYTE
+EXTRACT_INT_SMALL(B, 16, iRegINoSp, smov, B)
+EXTRACT_INT_LARGE(B, 16, iRegINoSp, T_BYTE)
+
+// SHORT
+EXTRACT_INT_SMALL(S, 8, iRegINoSp, smov, H)
+EXTRACT_INT_LARGE(S, 8, iRegINoSp, T_SHORT)
+
+// INT
+EXTRACT_INT_SMALL(I, 4, iRegINoSp, umov, S)
+EXTRACT_INT_LARGE(I, 4, iRegINoSp, T_INT)
+
+// LONG
+EXTRACT_INT_SMALL(L, 2, iRegLNoSp, umov, D)
+EXTRACT_INT_LARGE(L, 2, iRegLNoSp, T_LONG)
+
+dnl
+dnl EXTRACT_FP($1,   $2,   $3,    $4,   $5   )
+dnl EXTRACT_FP(type, insn, index, size, shift)
+define(`EXTRACT_FP', `
+instruct extract$1(vReg$1 dst, vReg src, immI idx) %{
+  match(Set dst (Extract$1 src idx));
+  effect(TEMP_DEF dst);
+  format %{ "extract$1 $dst, $src, $idx" %}
+  ins_encode %{
+    int index = (int)$idx$$constant;
+    if (index == 0) {
+      __ $2($dst$$FloatRegister, $src$$FloatRegister);
+    } else if (index < $3) {
+      __ ins($dst$$FloatRegister, __ $4, $src$$FloatRegister, 0, index);
+    } else {
+      assert(UseSVE > 0, "must be sve");
+      __ sve_orr($dst$$FloatRegister, $src$$FloatRegister, $src$$FloatRegister);
+      __ sve_ext($dst$$FloatRegister, $dst$$FloatRegister, index << $5);
+    }
+  %}
+  ins_pipe(pipe_slow);
+%}')dnl
+dnl
+// FLOAT
+EXTRACT_FP(F, fmovs, 4, S, 2)
+
+// DOUBLE
+EXTRACT_FP(D, fmovd, 2, D, 3)
+
+// ------------------------------ Vector mask loat/store -----------------------
```

Comment:
Should be "load/store"

Reply:
Yes, it's a typo. Updated in the new revision.